transcom/mymove
# B 18428 SSW Form PDF Work - INT

## [B-18428](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A883374)

## Summary
Note: This is dependent on [B-18429](https://github.com/transcom/mymove/pull/11735), which has just been reconfigured for Integration Testing.
The Shipment Summary Worksheet is a file generated as part of the PPM Process. This task was to replace the existing png template images with a correctly filling PDF form template. This template can be viewed in Agility attachments for this story. This template was created in adobe, and then programmatically filled with a new service, SSWPPMGenerator. All new functions have unit tests written. A command and an API have both been updated to reflect the success of the generator, and will be elaborated on in testing instructions.


## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

There are three tests that should be performed if possible, ranked from most to least important:

**1. Access an SSW for an existing PPM Shipment. Requires a PPMShipment UUID.**
The internal API Endpoint is: /moves/{PPMShipmentID}/shipment_summary_worksheet?preparationDate=yyyy-mm-dd
So, an example would be:
http://milmovelocal:3000/internal/moves/f093a13b-4ab8-4545-b24c-eb44bf52e605/shipment_summary_worksheet?preparationDate=2024-01-24
If you are logged in to MilMove as a User, and paste it into your URL with a correct PPMShipmentID from the database, and a date, it should return the PDF to your browser. 
**This is also testable in SwaggerUI, but it will return PDF Code in the response viewer instead of to your browser, so please use the url bar.**
Additionally, please be aware that many fields added in this version of the SSW are not yet available in the application. For a complete list of values and why they are unfilled, please review the Agility story shortly.

**2. Run the SSW Server test package, ensure all tests pass**
These tests will test all new functions, as well as the form filling. It will fail if there is a corrupted or incorrectly configured PDF.

**3. Run the "go run ./cmd/generate-shipment-summary" command.**
When this command is ran, the last two pieces of data returned in the console should be the tempfile name, and the number 2 respectively. The temp file name ensures the memory filesystem is working correctly, and the number 2 means that the PDF is not corrupted (2 is the pagecount of pdf info programmatically retrieved)

### Backend
